### PR TITLE
Rename `--only-deps` to `--fetch-deps-only`, make it only populate MOVE_HOME

### PR DESCRIPTION
**Pull request by mkurnikov**
## Motivation

Fixes https://github.com/move-language/move/issues/236

Overwrites https://github.com/move-language/move/pull/104

AFAIK, the only IDE that uses dependencies source code is the Intellij-move plugin. Move-analyzer yet to support that, I assume no breakages there. 

@vladimirovmm @awelc @tnowacki 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?
Yes

## Test Plan

Testing would require mocking `MOVE_HOME`, calling the command with the flag, and checking that correct git repos are cloned. I want to hear the feedback first, then I'll try to write one.

### CODE REVIEW

File: language/tools/move-package/src/resolution/resolution_graph.rs [hunk 1]
```diff
@@ -515,6 +515,33 @@ impl ResolvingGraph {
...
+        // include dev dependencies if in dev mode
+        let additional_deps = if build_options.dev_mode {
+            package.dev_dependencies.clone()
+        } else {
+            BTreeMap::new()
+        };
+
+        for (dep_name, dep) in package
+            .dependencies
+            .into_iter()
```
Comment (tnowacki) [inline language/tools/move-package/src/resolution/resolution_graph.rs:532]: should this just be an `iter()`? Then everything else here can be by reference?

**Review by tnowacki — COMMENTED** [review]
No issue with the change to the flag. Just some comments on the implementation and then should be good to go :)

File: language/tools/move-package/src/resolution/resolution_graph.rs (hunk 1 above)
Comment (mkurnikov) [reply language/tools/move-package/src/resolution/resolution_graph.rs]: I've updated the code, hopefully it's what you had in mind.

**Comment by mkurnikov** [comment]
On the test side of things, it's a network-heavy feature that requires mocking many things. Considering its niche use case, I'd leave it untested.

File: language/tools/move-package/src/resolution/resolution_graph.rs (hunk 1 above)
Comment (mkurnikov) [reply language/tools/move-package/src/resolution/resolution_graph.rs]: Yes, but in 
```
        if config.fetch_deps_only {
            config.download_deps_for_package(&rerooted_path)?;
            return Ok(());
        }
```
https://github.com/move-language/move/pull/267/files#diff-5d804e276a879688a14ffe2fc7342accb0050d6e84418a8cd855a9faa93d1ce4R18
`config` is destroyed after the `download_deps_for_package` anyway, so it doesn't need to be a reference. 

Same for the `SourceManifest`, it's not used outside the `download_dependency_repos` call, so there's no need for a reference. Unless I'm missing something. 

I've fixed the `root_path`, it does remove one `clone()`.

Comment (tnowacki) [reply language/tools/move-package/src/resolution/resolution_graph.rs]: > config is destroyed after the download_deps_for_package anyway

Sure, but you aren't actually using it by-value in any meaningful or required way. So if anyone needs to modify surrounding code afterwards, they are now stuck.

It is best to use things by-reference unless you have some reason not to (prevents clones, ensures proper mutations, etc)

Comment (mkurnikov) [reply language/tools/move-package/src/resolution/resolution_graph.rs]: Could you review it again? I made everything into a by-reference, and extracted the only mutation required for the `mut self` to the higher scope.

File: language/tools/move-package/src/resolution/resolution_graph.rs [hunk 2]
```diff
@@ -515,6 +515,30 @@ impl ResolvingGraph {
...
     }
 
+    pub fn download_dependency_repos(
+        manifest: &SourceManifest,
+        build_options: &BuildConfig,
+        root_path: &Path,
+    ) -> Result<()> {
+        // include dev dependencies if in dev mode
+        let additional_deps = if build_options.dev_mode {
+            manifest.dev_dependencies.clone()
```
Comment (tnowacki) [inline language/tools/move-package/src/resolution/resolution_graph.rs:525]: This clone isn't needed if you are just using `.iter()` later.

I imagine you did this `clone` due to Rust borrow checker issues, and it is indeed an annoying one. 

You might need to do this:
```
let empty_deps;
let additional_deps = if build_options.dev_mode {
        manifest.dev_dependencies.clone()
} else {
    empty_deps = BTreeMap::new();
    &empty_deps
};
```

File: language/tools/move-package/src/resolution/resolution_graph.rs [hunk 3]
```diff
@@ -515,6 +515,30 @@ impl ResolvingGraph {
...
+        } else {
+            BTreeMap::new()
+        };
```
Comment (tnowacki) [inline language/tools/move-package/src/resolution/resolution_graph.rs:528]:
```suggestion
        let empty_deps;
        let additional_deps = if build_options.dev_mode {
            manifest.dev_dependencies.clone()
        } else {
            empty_deps = BTreeMap::new();
            &empty_deps
        };
```
If you just want to apply it here :)

**Review by tnowacki — COMMENTED** [review]
Everything looks good otherwise! Sorry for the clone pedantry, I would normally just land it and make any changes as needed later, but I don't have a ton of time for that sort of thing these days unfortunately.

**Review by tnowacki — APPROVED** [review]
Sorry for being such a curmudgeon, thanks for the PR!

**Pull request merged**